hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -96,6 +96,12 @@ public static void cascade(
 					if ( hasUninitializedLazyProperties && persister.getPropertyLaziness()[i] && action.performOnLazyProperty() ) {
 						LazyAttributeLoadingInterceptor interceptor = persister.getInstrumentationMetadata().extractInterceptor( parent );
 						child = interceptor.fetchAttribute( parent, propertyName );
```

Comment:
I could definitely be missing something here, but it looks to me like `$$_hibernate_read_details` would do what is needed for both lazy and extra-lazy collections. 

For lazy collections, `$$_hibernate_read_details` would instantiate the `PersistentCollection`, load its elements, and assign the `PersistentCollection` into the owner entity.

For extra-lazy collections,  `$$_hibernate_read_details` would instantiate the `PersistentCollection` and assign the `PersistentCollection` into the owner entity. The `PersistentCollection` size would be cached, but its elements would not be loaded.

I tried calling `EntityPersister#getPropertyValue(Object object, String propertyName)`, thinking that would end up calling  `$$_hibernate_read_details`, but it doesn't. Instead assigns an empty collection. I believe this is the cause of the bug.

How should Hibernate call `$$_hibernate_read_details`, via `EntityPersister#getPropertyValue` (this would need to be fixed), or some other way. 

How can Hibernate explicitely call `$$_hibernate_read_details`?